config.m4

```diff
@@ -406,6 +406,12 @@ if test "$PHP_MONGODB_SASL" != "no"; then
   ], [
     -L$MONGODB_SASL_DIR/$PHP_LIBDIR
   ])
+
+  PHP_CHECK_LIBRARY(sasl2, sasl_client_done,
+  [
+    MONGOC_HAVE_SASL_CLIENT_DONE=1
+    AC_SUBST(MONGOC_HAVE_SASL_CLIENT_DONE)
+  ])
```

Comment:
I noticed elsewhere in `config.m4` we split the constant definition and `AC_SUBST` into separate statements. Is there any reason we don't simplify as `AC_SUBST(MONGOC_HAVE_SASL_CLIENT_DONE, 1)`?